src/riot-rs-threads/src/arch/mod.rs

```diff
@@ -1,17 +1,65 @@
+/// Arch-specific implementations for the scheduler.
+pub trait Arch {
+    const DEFAULT_THREAD_DATA: Self::ThreadData;
```

Comment:
I guess this is not using `Default` b/c we need to be able to use this in `const` context?

Reply:
Unfortunately yes.